File: Backend/MD_CRM_CRUD_JWT_Auth/Responses/APIResponse.cs
using System.Net;

namespace MD_CRM_CRUD_JWT_Auth.Responses
{
    public class APIResponse
    {
        public HttpStatusCode httpStatusCode { get; set; }
        public bool IsSuccess { get; set; } = true;
        public List<string> ErrorMessages { get; set; } = null;
        public object? Result { get; set; } = null;
    }
}

File: Frontend/src/app/models/APIResponse.ts

export class APIResponse {
    public httpStatusCode: any;
    public isSuccess: boolean = true;
    public errorMessages: string[] | null = null;
    public result: any | null = null;

    constructor(
        httpStatusCode: any,
        MyProperty: number,
        IsSuccess: boolean = true,
        ErrorMessages: string[] | null = null,
        Result: any | null = null
    ) {
        this.httpStatusCode = httpStatusCode;
        this.isSuccess = IsSuccess;
        this.errorMessages = ErrorMessages;
        this.result = Result;
    }
}
